### PR TITLE
[PFSP1-0003] Register Screen

### LINKED ISSUE #22: [PFSP1-0003] Register Screen
### Priority 
- [ ] Low
- [ ] Mid
- [x] High

### Links
Name | Link
-|-
User Story | [TBD]()
Figma | [link](https://www.figma.com/file/YWJimHMl9fq3v4dQBZSuQo/Hi-Fi-User?node-id=71%3A26)

### Completion
Expected | Time
-|-
Expected Start | 29-07-2020
Expected End | 03-08-202

### PR DESCRIPTION
**Pull request by robbyaww**
[PFSP1-0003] Register Screen
Closes #22 
***
## Description
Add register page

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [link]()
Figma | [link]()
Issue | [link]()

## Todos
- [x] UI
- [x] Coordinator
- [x] Network
- [ ] Tests
- [ ] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.

### UI Changes
UI related changes will be shown here (can be either screenshot or video format)

### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
Additional notes that you have either technical debt, blocker, etc.

### CODE REVIEW

File: MainApp/MainApp/Feature/Register/Register.xcassets/ic_google.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (robbyaww) [inline MainApp/MainApp/Feature/Register/Register.xcassets/ic_google.imageset/Contents.json:1]: add register.xcasset file

File: MainApp/MainApp/Feature/Register/Register.xcassets/ic_google.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "filename" : "image 1@3x.png",
+      "idiom" : "universal",
+      "scale" : "3x"
+    }
+  ],
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
Comment (robbyaww) [inline MainApp/MainApp/Feature/Register/Register.xcassets/ic_google.imageset/Contents.json:23]: add register.xcasset file

File: MainApp/MainApp/Feature/Register/RegisterViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return emailPred.evaluate(with: email)
+    }
+    
+    func isValidPassword(_ password: String) -> Bool {
+        let PasswordRegEx = "^(?=.*[A-Z].*[A-Z])(?=.*[!@#$&*])(?=.*[0-9].*[0-9])(?=.*[a-z].*[a-z].*[a-z]).{8}$"
+        
+        let passwordPred = NSPredicate(format:"SELF MATCHES %@", PasswordRegEx)
+        return passwordPred.evaluate(with: password)
+    }
+}
```
Comment (robbyaww) [inline MainApp/MainApp/Feature/Register/RegisterViewModel.swift:35]: already moved logic to viewModel

File: MainApp/MainApp/Network/Path/FeaturePath/Register/RegisterNetwork.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
+//
+//  RegisterNetwork.swift
+//  MainApp
+//
+//  Created by Robby Awalul Meviansyah Abdillah on 09/08/20.
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import Foundation
+import RxSwift
```
Comment (inxudianz) [inline MainApp/MainApp/Network/Path/FeaturePath/Register/RegisterNetwork.swift:10]: dont import RX if not used

**Review by inxudianz — DISMISSED** [review]
next time dont create copy branch, fix issue in original branch

**Review by inxudianz — APPROVED** [review]

**Pull request merged**